File: backend/app/schemas/finding.py
from pydantic import Field

from app.models.finding import FindingSeverity, FindingStatus
from app.schemas.base import BaseSchema


class FindingSummary(BaseSchema):
    id: str
    project_id: str
    scan_id: str
    asset_id: str
    title: str
    description: str | None = None
    severity: FindingSeverity
    status: FindingStatus


class FindingListResponse(BaseSchema):
    items: list[FindingSummary]
    total: int


class UpdateFindingRequest(BaseSchema):
    status: FindingStatus | None = None
    severity: FindingSeverity | None = None
    title: str | None = Field(default=None, min_length=1, max_length=512)
    description: str | None = Field(default=None, max_length=10000)

File: frontend/src/shared/types/finding.ts
export type FindingSeverity = "critical" | "high" | "medium" | "low" | "info";
export type FindingStatus =
  | "open"
  | "in_review"
  | "accepted"
  | "resolved"
  | "false_positive";

export interface FindingSummary {
  id: string;
  project_id: string;
  asset_id: string;
  scan_id?: string | null;
  source?: string;
  finding_code?: string | null;
  title: string;
  description?: string | null;
  severity: FindingSeverity;
  status: FindingStatus;
  risk_score?: number | null;
  detected_at?: string | null;
  created_at?: string;
  updated_at?: string;
}

export interface FindingListQuery {
  page?: number;
  limit?: number;
  status_group?: "" | "open" | "resolved" | "ignored";
  severity?: FindingSeverity | "";
  search?: string;
  sort?: "risk_score" | "severity" | "title" | "created_at";
  order?: "asc" | "desc";
}

export interface FindingListData {
  items: FindingSummary[];
  total: number;
  page: number;
  limit: number;
}

export interface UpdateFindingRequest {
  status?: FindingStatus;
}
